tests/integration/Core/Content/Category/SalesChannel/CategoryRouteTest.php

```diff
@@ -139,11 +162,207 @@ public function testHomeWithSalesChannelOverride(): void
+    }
+
+    /**
+     * @return CmsInheritanceDataProviderIterator
+     */
+    public static function categoryCmsPageInheritanceDataProvider(): iterable
+    {
+        yield from CategoryRouteInheritanceFixtures::noOverridesDataProvider();
+        yield from CategoryRouteInheritanceFixtures::mixedOverridesDataProvider();
+        yield from CategoryRouteInheritanceFixtures::allOverridesDataProvider();
+
+        yield from CategoryRouteInheritanceFixtures::enOverridesDataProvider();
+        yield from CategoryRouteInheritanceFixtures::deOverridesDataProvider();
+        yield from CategoryRouteInheritanceFixtures::atOverridesDataProvider();
+    }
+
+    /**
+     *  Testing CMS slot inheritance with category overrides.
+     *  - EN is System Default language
+     *  - DE is parentless
+     *  - AT inherits from DE
+     *  Expected hierarchy: Overrides of categories (AT > DE > EN) > Templates of categories (AT > DE > EN)
+     *
+     * @param CmsInheritanceDataProviderActual $actual
+     */
+    #[DataProvider('categoryCmsPageInheritanceDataProvider')]
+    public function testCategoryCmsPageInheritance(array $actual, string $expectedContentValue): void
+    {
+        $context = Context::createDefaultContext();
+        $this->createLanguages($context);
+        $this->createTranslatedData($actual, $context);
+        $this->createDomains($context);
+
+        $this->browser->request(
+            'POST',
+            '/store-api/category/' . $this->ids->get('category'),
+            [],
+        );
+
+        $this->assertLandingPageCmsPage(
+            $this->ids->get('category'),
+            $this->ids->get('cms-page'),
+            $expectedContentValue,
+        );
+    }
+
+    /**
+     * @return CmsInheritanceDataProviderIterator
+     */
+    public static function categoryCmsPageInheritanceWithDuplicatesInLanguageChainDataProvider(): iterable
+    {
+        yield from CategoryRouteInheritanceFixtures::duplicatesInLanguageChainDataProviderNoOverrides();
+        yield from CategoryRouteInheritanceFixtures::duplicatesInLanguageChainDataProviderEnOverrides();
+        yield from CategoryRouteInheritanceFixtures::duplicatesInLanguageChainDataProviderDeOverrides();
+        yield from CategoryRouteInheritanceFixtures::duplicatesInLanguageChainDataProviderAllOverrides();
+    }
+
+    /**
+     *  Testing CMS slot inheritance with category overrides, with multiple EN occurances in language chain.
+     *  - EN is System Default language
+     *  - DE inherits from EN
+     *  Expected hierarchy: Overrides of categories (DE > EN) > Templates of categories (DE > EN)
+     *  Actual language chain hierarchy: DE > EN > EN
+     *
+     * @param CmsInheritanceDataProviderActual $actual
+     */
+    #[DataProvider('categoryCmsPageInheritanceWithDuplicatesInLanguageChainDataProvider')]
+    public function testCategoryCmsPageInheritanceWithDuplicatesInLanguageChain(array $actual, string $expectedContentValue): void
```

Comment:
You test a lot of cases via integration tests. This can affect our integration runtime duration. Do you checked how long it takes to run these tests? You should use integration tests for the general process and cover all edge cases with unit tests imo.

Reply:
Locally like 5 seconds for all 63 cases. But of course, you have point there. I'll have a look how other route tests do unit tests 😄

Reply:
5 seconds are a lot for a single test class 🤔 we should not forget that these 5 seconds are on your fast machine and not on our github runner. The whole magic appears in the `buildCmsSlotConfig ` function, right? You could write unit tests and cover all edge cases with unit tests. This would be much faster, because you would not insert data to the database and send requests in each single run. For the integration part i would keep only few specific cases if the inheritance works in general :)

Reply:
Sounds good 👍

Reply:
Reduced it from to ~3.5s, I'd like to keep the rest of the tests, so that each case is still adequately covered. Hope that's fine 👍